raiden/utils/logging.py

```diff
@@ -1,12 +1,15 @@
+from copy import deepcopy
+
 from raiden.utils.typing import Dict
 
 
 def redact_secret(data: Dict) -> Dict:
-    """Modify `data` in-place and replace keys named `secret`."""
+    """Modify `data` and replace keys named `secret`."""
     if not isinstance(data, dict):
         raise ValueError("data must be a dict.")
 
-    stack = [data]
+    data_copy = deepcopy(data)
```

Comment:
Can you add a `FIXME: assess performance impact of this deepcopy` here?

Reply:
✅